sdk/eventhub/azure-eventhubs/azure/eventhub/producer.py

```diff
@@ -370,3 +229,15 @@ def close(self, exception=None):
         else:
             self.error = EventHubError("This send handler is now closed.")
         self._handler.close()
+
+
+def _error(outcome, condition):
+    if outcome != constants.MessageSendResult.Ok:
+        raise condition
+
+
+def _set_partition_key(event_datas, partition_key):
+    ed_iter = iter(event_datas)
+    for ed in ed_iter:
+        ed._set_partition_key(partition_key)
+        yield ed
```

Comment:
Just a stylistic thing - but I like to put internal functions like these at the top of the file to make them easy to find :)

Reply:
good suggestion. Fixed